### ISSUE TITLE
scanner can't bind hci0

### ISSUE THREAD

**Issue by efidoman** [go-ble/ble#20]
I'm getting an error running the scanner example app as root on debian:

Is it not possible to have other unix processes connected to hci at the same time as this library? I'm trying to build an application that just scans and connects to periphials, but there are other apps using the hci0 as well for connecting to other periphials via bluetooth as well.

2018/04/20 07:29:59 can't new device : can't init hci: no devices available: (hci0: can't bind socket to hci user channel: invalid argument)

**Comment by schollz** [comment]
@efidoman Did you figure this error out?

I'm also getting it. It seems somehow this library (and the one it forks from) causes a power down of the bluetooth device and I can't get it to turn back on until I do a restart.

I got the error `can't init hci: no devices available: (hci0: can't up device: connection timed out)`. So then I tried scanning the regular way from the command line and saw

```
%  sudo btmgmt find
Unable to start discovery. status 0x0f (Not Powered)
```

Doing a `sudo service bluetooth restart` doesn't fix it. Only a reboot (or sometimes two) seems to fix it.
